diff --git a/haskell-src/Concordium/Types/ProtocolLevelTokens/CBOR.hs b/haskell-src/Concordium/Types/ProtocolLevelTokens/CBOR.hs
index f0771a21..9963ebaa 100644
--- a/haskell-src/Concordium/Types/ProtocolLevelTokens/CBOR.hs
+++ b/haskell-src/Concordium/Types/ProtocolLevelTokens/CBOR.hs
@@ -594,6 +594,10 @@ encodeAccountAddress (AccountAddress (FBS.FixedByteString ba)) =
 tokenAccountAddressFromBytes :: LBS.ByteString -> Either String AccountAddress
 tokenAccountAddressFromBytes = decodeFromBytes decodeAccountAddress "token account address"
 
+-- | Encode an 'AccountAddress' to a strict bytestring.
+tokenAccountAddressToBytes :: AccountAddress -> BS.ByteString
+tokenAccountAddressToBytes = CBOR.toStrictByteString . encodeAccountAddress
+
 -- | A destination that can receive and hold protocol-level tokens.
 --  Currently, this can only be a Concordium account address.
 data TokenHolder = HolderAccount
